### ISSUE TITLE
"IO error: stream did not contain valid UTF-8" even though the file is encoded as UTF-8

### ISSUE THREAD

**Issue by bastilimbach** [BrettMayson/HEMTT#654]
**Tools (complete and add to the following information):**
- **HEMTT:** `1.11.1`
- **Operating System:** Ubuntu 22.04.4 LTS (WSL) & GitHub Actions enviroment (ubuntu-latest)

**Description:**
This one is very strange... After calling `hemtt release` or `hemtt build` I *sometimes* get the following error:
```bash
ERROR Preprocessor error: Workspace Error: VfsError: Could not read path for '/addons/main/CfgPatches.cpp': IO error: stream did not contain valid UTF-8
```

If I now run the same command several times **without changing any files**, it succeeds. I can spam the build or release command for 10 seconds and after several errors sometimes it just works. Again, without changing any files.

I also double-checked the encoding. All files are UTF-8 encoded.

**Steps to reproduce:**
I created a GitHub repository to reproduce the bug. You can check out the GitHub Action failure here: https://github.com/bastilimbach/hemtt-io-error/actions/runs/7951811810/job/21705671274#step:4:53

Or reproduce it locally:
1. `git clone https://github.com/bastilimbach/hemtt-io-error.git`
2. `cd hemtt-io-error `
3.  run `hemtt release` or `hemtt build` several times until it eventually succeeds.

**Expected behavior:**
Always have the build succeed.

**Screen capture of the behavior:**
![hemtt](https://github.com/BrettMayson/HEMTT/assets/5406078/e92eba00-51da-4db8-9b5a-1022b348dc60)
> This is not speeded up. I didn't change any files. Simply pressed "up" and "enter" in the terminal.

**Comment by jokoho48** [comment]
if you change the 2 files CfgPatches.cpp and CfgFunctions.cpp to hpp it works as expected as it should

**Comment by BrettMayson** [comment]
This has been fixed in 78e0f2f23f104b5cc111b258e1a0d99e634fd667, with a warning. `config.cpp` is the really the only thing that should have `.cpp`, Includes should be using `.hpp`

What was happening, is it was rapifying your `CfgFunctions.cpp` and `CfgPatches.cpp`, then when it got to `config.cpp` and tried to include them, they were now binary blobs. Just by the nature of multithreading, occasionally it would do `config.cpp` first / fast enough to read the original data before they were rapified.
